components/script/dom/document.rs

```diff
@@ -2359,8 +2359,11 @@ impl DocumentMethods for Document {
     }
 
     // https://html.spec.whatwg.org/multipage/#dom-document-location
-    fn Location(&self) -> Root<Location> {
-        self.location.or_init(|| Location::new(&self.window))
+    fn GetLocation(&self) -> Option<Root<Location>> {
+        match self.browsing_context() {
+            Some(browsingcontext) => return Some(self.location.or_init(|| Location::new(&self.window))),
+            None => return None,
+        }
```

Comment:
Lame attempt at clarifying:

If you want the variant to go from `Some` → `Some` use `map`

If you want the variant to go from `Some` → (`Some` or `None`) use `and_then`